### DISCUSSION TITLE
Developer tools

### DISCUSSION THREAD

**Discussion by stephen-lev** [OP-TED/eForms-SDK#914] [Q&A] [answered]
Hi!

I'm keen to learn about the developer tools utilized in the development of the SDK. Specifically, I'm interested in the IDEs, plugins, or extensions that facilitate efficient editing of various file types such as *.xsd, *.sch, *.xml, *.gc, *.json, and others within a cohesive project environment. Could you kindly share your recommendations and experiences in this regard?

**Comment by rouschr** [comment]
Hello, 

Visual Studio Code is quite good for basic support on many extensions.
I sometimes use Eclipse to edit xml, but as we generate most SDK files we rarely edit them directly.

**Reply by stephen-lev** [reply]
Thanks for the quick answer!

Could you elaborate on the tools or processes you utilize for this generation? Also what kind of modelling tool are you using for creating the model on which the generation is based on? Understanding the source and workflow behind SDK file generation would provide valuable insights into your development process. Any details you can share would be much appreciated.

**Reply by rouschr** [reply, marked as answer]
As the whole thing is about managing data we use a classic relational database, except for the "schemas" which we maintain "by hand" for now. For input we have a mix of data coming from custom web apps and imports from a custom command line tool, so the data comes from multiple people (after discussions, ...).
We read from the database using a Java command line tool to export/generate the SDK JSON, XML and various other files.
At the end we also have a tool to check the generated content using rules, for that we use Java Drools.
I hope this gives you some ideas.

**Reply by stephen-lev** [reply]
Thanks for the answer!
